### PR TITLE
[16.0][IMP] base_tier_validation: Add _prepare_tier_review_vals() methot to allow inheritance

### PR DESCRIPTION
**Pull request by victoralmau**
FWP from 15.0: https://github.com/OCA/server-ux/pull/762

Add `_prepare_tier_review_vals()` methot to allow inheritance.

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT45804

### CODE REVIEW

File: base_tier_validation/models/tier_validation.py [hunk 1]
```diff
@@ -472,15 +481,8 @@ def request_validation(self):
...
-                                {
-                                    "model": self._name,
-                                    "res_id": rec.id,
-                                    "definition_id": td.id,
-                                    "sequence": sequence,
-                                    "requested_by": self.env.uid,
-                                }
-                            )
+                            vals = rec._prepare_tier_review_vals(td, sequence)
+                            created_trs += tr_obj.create(vals)
```
Comment (pedrobaeza) [inline base_tier_validation/models/tier_validation.py]: Please take the occasion to accumulate the dictionaries and create all the records at once.

File: base_tier_validation/models/tier_validation.py [hunk 2]
```diff
@@ -469,18 +478,12 @@ def request_validation(self):
                         order="sequence desc",
                     )
                     sequence = 0
+                    vals = []
```
Comment (pedrobaeza) [inline base_tier_validation/models/tier_validation.py]: For following naming conventions:

```suggestion
                    vals_list = []
```

Comment (pedrobaeza) [reply base_tier_validation/models/tier_validation.py]: And declare it outside the loop and do only one create.

Comment (pedrobaeza) [reply base_tier_validation/models/tier_validation.py]: This last comment is unattended.

Comment (victoralmau) [reply base_tier_validation/models/tier_validation.py]: It's changed

Comment (pedrobaeza) [reply base_tier_validation/models/tier_validation.py]: Outside the other loop, and you can join the 2 `if` lines into one.

File: base_tier_validation/models/tier_validation.py [hunk 3]
```diff
@@ -472,16 +482,11 @@ def request_validation(self):
...
-                                    "res_id": rec.id,
-                                    "definition_id": td.id,
-                                    "sequence": sequence,
-                                    "requested_by": self.env.uid,
-                                }
+                            vals_list.append(
+                                rec._prepare_tier_review_vals(td, sequence)
                             )
                     self._update_counter({"review_created": True})
+        created_trs += tr_obj.create(vals_list)
```
Comment (pedrobaeza) [inline base_tier_validation/models/tier_validation.py]: No need now of pre-setting the variable:

```suggestion
        created_trs = tr_obj.create(vals_list)
```

(and remove its initialization).

And as said, you can also reduce one indentation level joining the 2 if.

**Review by pedrobaeza — APPROVED** [review]
Thanks for attending all the nitpickings.

**Review by LoisRForgeFlow — APPROVED** [review]
Thanks!

/ocabot merge patch

**Pull request merged**